File: src/lib/veggies.ts
import type { VegType } from '@/components/VegIcon';

// The collection: you start with the tomato (the pomodoro itself) and unlock a
// new vegetable as your all-time pomo count grows.

export interface Veg {
  type: VegType;
  name: string;
  unlockAt: number; // all-time pomos required
}

export const VEGGIES: Veg[] = [
  { type: 'tomato', name: 'Tomato', unlockAt: 0 },
  { type: 'carrot', name: 'Carrot', unlockAt: 10 },
  { type: 'strawberry', name: 'Strawberry', unlockAt: 25 },
  { type: 'pepper', name: 'Pepper', unlockAt: 50 },
  { type: 'corn', name: 'Corn', unlockAt: 80 },
  { type: 'eggplant', name: 'Eggplant', unlockAt: 120 },
  { type: 'broccoli', name: 'Broccoli', unlockAt: 180 },
  { type: 'mushroom', name: 'Mushroom', unlockAt: 260 },
];

export function isUnlocked(veg: Veg, totalPomos: number): boolean {
  return totalPomos >= veg.unlockAt;
}

export function unlockedCount(totalPomos: number): number {
  return VEGGIES.filter((v) => isUnlocked(v, totalPomos)).length;
}

/** The next vegetable to unlock, and how many pomos remain — or null at the end. */
export function nextVeg(totalPomos: number): { veg: Veg; remaining: number } | null {
  const next = VEGGIES.find((v) => totalPomos < v.unlockAt);
  if (!next) return null;
  return { veg: next, remaining: next.unlockAt - totalPomos };
}

File: PomoLeagueApp/PomoLeague/Models/Veggies.swift
import Foundation

enum VegType: String, CaseIterable, Identifiable {
    case tomato, carrot, strawberry, pepper, corn, eggplant, broccoli, mushroom
    var id: String { rawValue }
    var name: String { rawValue.capitalized }
}

struct Veg: Identifiable {
    var type: VegType
    var unlockAt: Int
    var id: String { type.rawValue }
    var name: String { type.name }
}

enum Veggies {
    static let all: [Veg] = [
        Veg(type: .tomato, unlockAt: 0),
        Veg(type: .carrot, unlockAt: 10),
        Veg(type: .strawberry, unlockAt: 25),
        Veg(type: .pepper, unlockAt: 50),
        Veg(type: .corn, unlockAt: 80),
        Veg(type: .eggplant, unlockAt: 120),
        Veg(type: .broccoli, unlockAt: 180),
        Veg(type: .mushroom, unlockAt: 260),
    ]

    static func isUnlocked(_ veg: Veg, total: Int) -> Bool { total >= veg.unlockAt }
    static func unlockedCount(total: Int) -> Int { all.filter { total >= $0.unlockAt }.count }

    static func next(total: Int) -> (veg: Veg, remaining: Int)? {
        guard let veg = all.first(where: { total < $0.unlockAt }) else { return nil }
        return (veg, veg.unlockAt - total)
    }
}
